modules/library/coverage/src/test/java/org/geotools/coverage/grid/GridCoverageTest.java

```diff
@@ -72,4 +76,40 @@ public void testSerialization() throws IOException, ClassNotFoundException {
             assertRasterEquals(coverage, serial);
         }
     }
+
+    @Test
+    public void testSubSampling() throws IllegalArgumentException, IOException, TransformException {
```

Comment:
I would add a couple more tests:

- Overviews are supported and enabled, check the right one gets picked
- Overviews are supported and enabled, but the last overview is still much bigger than the requested image, allowing for both overview usage and subsampling on top of it, and checking the subsampling factor is "right" for the overview, and not computed on the base image.